src/IO/H5/Python/File.cpp

```diff
@@ -17,32 +17,73 @@ void bind_h5file(py::module& m) {  // NOLINT
   // Wrapper for basic H5File operations
   using h5_file_rw = h5::H5File<h5::AccessType::ReadWrite>;
   py::class_<h5_file_rw>(m, "H5File")
-      .def(py::init<std::string, bool>(), py::arg("file_name"),
-           py::arg("append_to_file") = false)
+      .def(py::init([](const std::string& file_name, bool append_to_file) {
+             const bool file_exists =
+                 file_system::check_if_file_exists(file_name);
+             if (file_exists && !append_to_file) {
+               throw std::runtime_error("File " + file_name +
```

Comment:
Please add a comment somewhere above that briefly explains why you add this error handling code, and link the issue https://github.com/sxs-collaboration/spectre/issues/2312